seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/splitter/AbstractJdbcSourceChunkSplitter.java

```diff
@@ -379,12 +383,41 @@ protected SnapshotSplit createSnapshotSplit(
     protected Column getSplitColumn(
             JdbcConnection jdbc, JdbcDataSourceDialect dialect, TableId tableId)
             throws SQLException {
-        Optional<PrimaryKey> primaryKey = dialect.getPrimaryKey(jdbc, tableId);
         Column splitColumn = null;
+        Table table = dialect.queryTableSchema(jdbc, tableId).getTable();
+
+        // first , compare user defined split column is in the primary key or unique key
+        Map<String, String> splitColumnsConfig = new HashMap<>();
+        try {
+            splitColumnsConfig = sourceConfig.getSplitColumn();
+        } catch (Exception e) {
+            log.error("Config snapshot.split.column get exception in {}:{}", tableId, e);
+        }
+        String tableSc =
+                splitColumnsConfig.getOrDefault(tableId.catalog() + "." + tableId.table(), null);
+
+        if (StringUtils.isNotEmpty(tableSc)) {
+            boolean isUniqueKey = dialect.isUniqueKey(jdbc, tableId, tableSc);
```

Comment:
```suggestion
            boolean isUniqueKey = dialect.getUniqueKeys(jdbc, tableId).contains(tableSc);
```


https://github.com/apache/seatunnel/blob/d2576ff13112b0f845930374632a959cf29e7eb9/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/dialect/JdbcDataSourceDialect.java#L112

Reply:
cc @Hisoka-X

Reply:
+1

Reply:
The above is pseudo code

this method `getUniqueKeys ` return `List<ConstraintKey>`...

Reply:
@XenosK

Reply:
> The above is pseudo code
> 
> this method `getUniqueKeys ` return `List<ConstraintKey>`...

done @hailin0